### ISSUE TITLE
ParseException:SQLparser:42000!Query too complex: running out of stack space

### ISSUE THREAD

**Issue by yzchang** [MonetDB/MonetDB#7092]
**Describe the bug**
Just to document a low priority problem.  The attached query contains >86K ANDs and ORs. 

Run the file with `mclient`, you'll get:
```
ParseException:SQLparser:42000!Query too complex: running out of stack space
```
Don't know what the current limit is, but maybe we can improve the parser in the future to handle a larger number of WHERE conditions

**Software versions**
 - MonetDB version number [Oct2020, hg id: 5e7ef9c3f0bd]
 
[query-too-complex.sql.gz](https://github.com/MonetDB/MonetDB/files/6228661/query-too-complex.sql.gz)

**Comment by yzchang** [comment]
Added the query in `sql/test/BugTracker-2021/Tests/query-too-complex.Bug-7092`
@PedroTadim Should I mark it with `KNOWNFAIL`?

**Comment by PedroTadim** [comment]
There will always be a limit on the stack, so I can't consider this as a bug. We can improve however.

**Comment by PedroTadim** [comment]
You should always add the KNOWNFAIL remark on the stable branch.

**Comment by sjoerdmullender** [comment]
Also on the default branch since at some point that gets forked into a new stable branch.

**Comment by yzchang** [comment]
Added KNOWNFALL in Oct2020 and default.  Marked this as an enhancement issue.

**Comment by sreejithssharma** [comment]
is this resolved now? I still see this issue in 43.23 Jan 2022 SP5. Have 58K AND and OR conditions which is lesser than 86K mentioned earlier.

**Comment by njnes** [comment]
No extreemly large queries with huge number of AND and OR conditions are still an issue on a limited stack space. The current default posix threads stackspace is 2M on 64 bit systems. You could play with changing this compilation setting to handle larger queries. Having such extreem cases maybe a good moment to reconsider your query generation and/or database schema design.
